docs/developer-guide/writing-epics.md

```diff
@@ -0,0 +1,221 @@
+# Writing Epics
+
+Most of the asynchronous operation we are going to do in MapStore2 are implemented using epics. This guide help a developer to have the base concepts about epics and then provides the best practices to write and add your epics to mapstore.
+
+## Base Concepts
+
+**Epics** are the core element of the [redux](https://github.com/reduxjs/redux) middleware called [redux-observable](https://redux-observable.js.org/). **redux-observable** is based on **RxJS**.
+
+**RxJS** is a library for reactive programming using **Observables**, to make it easier to compose asynchronous or callback-based code.
+
+**stream** The concept of stream is *"sequence of data made available over time."*.
+
+```text
+--a---b-c---d---X---|->
+
+a, b, c, d are emitted values
+X is an error
+| is the 'completed' signal
+---> is the timeline
+
+```
+
+**Observable** is the core entity of RxJS and, more generically, of the whole reactive programming paradigm. Basically is an entity that emit events and can be subscribed, so then the subscribers can intercept the events emitted. It this the entity that implements the concept of **stream** (so *stream* and *Observable* are almost used as synonym).
+
+Subscribing observables can be hard, so the RxJs provides also a lot of **operators** that help to manipulate and combine observables (so, *streams*).
+here an example of how operators allow to manipulate event streams to count clicks:
+
+```text
+clickStream:    ---c----c--c----c------c--> <-- Stream of clicks
+                vvvvv map(c becomes 1) vvvv <-- opeartor that trasforms each event into a `1`
+                ---1----1--1----1------1--> <-- new stream returned by the operator
+                vvvvvvvvv scan(+) vvvvvvvvv <-- opeartor that does the sum
+counterStream:  ---1----2--3----4------5--> <-- click count stream returned by the operator
+```
+
+The final stream can be finally subscribed to update, for instance, a counter on the UI.
+
+### Versions
+
+At the time of writing this documentation MapStore2 are using RxJS 5.1.1 and redux-observable 0.13.0.
```

Comment:
I would not confuse the developer with too many details, just a note on the versions we use (and the link to the official documentation for those is enough in my opinion).

Reply:
Yes, but I'm afraid the original documentation will disappear soon, so I'd like to provide at least some basic information about how to read the actual code to posterity :).

Reply:
I hope the documentation of older versions will remain, maybe it will be more difficult to find them. If they are going to disappear, blame to the RxJS project. In any case, I am looking for a compromise: can we move this libraries-related docs to different page and link it from here?

Reply:
Hehe, I use to skip details I'm not interested, but if you think it's confusing, I'll remove it.